### PR TITLE
Bump rustc nightly-2022-11-22

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Bump rustc nightly-2022-08-30 and fix new errors and lints.

### Important Notes

* in a follow-up task, we will need to replace `websocket` which has a dependency (`traitobject` 0.1), that will stop compiling in a future version of rust. We currently get a warning to that effect. No upgrade of `websocket` itself is to be expected, as it is no longer updated. 

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

**Comment by mwu-tow** [comment]
> in a follow-up task, we will need to replace websocket which has a dependency (traitobject 0.1), that will stop compiling in a future version of rust. We currently get a warning to that effect. No upgrade of websocket itself is to be expected, as it is no longer updated.

A better maintained equivalent is `tungstenite` but due to somewhat different API switch would be really non-trivial. I once looked into this.
TBH now I hope that we will be able to drop this dependency altogether, once the parser replaces the Scala-based one.

**Review by mwu-tow — APPROVED** [review]

File: app/gui/controller/engine-protocol/src/language_server/types.rs
Comment (farmaazon) [inline app/gui/controller/engine-protocol/src/language_server/types.rs:111]: Why do we need to remove `Eq`?

Comment (MichaelMauderer) [reply app/gui/controller/engine-protocol/src/language_server/types.rs:111]: Because `ExpressionUpdatePayload` contains a variant that has a `f64` field.

**Review by farmaazon — APPROVED** [review]

**Comment by MichaelMauderer** [comment]
@JaroslavTulach @kazcw @wdanilo I need a review from a codeowner of the parser files.

File: lib/rust/ensogl/component/shadow/src/lib.rs
Comment (wdanilo) [inline lib/rust/ensogl/component/shadow/src/lib.rs]: Can we post a link to a report about the false positive please, so we will remove it after its fixed?

**Review by wdanilo — APPROVED** [review]
Amazing <3

File: lib/rust/ensogl/component/shadow/src/lib.rs
Comment (MichaelMauderer) [reply lib/rust/ensogl/component/shadow/src/lib.rs]: Turns out I missed something here and the borrow was needles after all. So, the `allow` is now removed (here and below).

**Pull request merged**
